docs/docs/development/hardware-integration/physical-layouts.md

```diff
@@ -0,0 +1,351 @@
+
+:::
+
+The `keys` property is required for [ZMK Studio](../../features/studio.md) support. It is used to describe the physical attributes of each key position present in that layout.
+To pull in the necessary definition for creating physical layouts with the `keys` property, a new include should be added to the top of the devicetree file:
+
+```
+#include <physical_layouts.dtsi>
+```
+
+Assigned to the `keys` property is an array of key descriptions listed in the same order as keymap bindings, matrix transforms, etc.
+A key description has the shape `<&key_physical_attrs w h x y r rx ry>` with the following properties:
+
+| Property   | Type     | Description                          | Unit                                                    |
+| ---------- | -------- | ------------------------------------ | ------------------------------------------------------- |
+| Width      | int (>0) | Key(cap) width                       | [centi-](https://en.wikipedia.org/wiki/Centi-)"keyunit" |
+| Height     | int (>0) | Key(cap) height                      | [centi-](https://en.wikipedia.org/wiki/Centi-)"keyunit" |
+| X          | uint     | Key X position (top-left point)      | [centi-](https://en.wikipedia.org/wiki/Centi-)"keyunit" |
+| Y          | uint     | Key Y position (top-left point)      | [centi-](https://en.wikipedia.org/wiki/Centi-)"keyunit" |
+| Rotation   | int      | Key rotation (positive => clockwise) | [centi-](https://en.wikipedia.org/wiki/Centi-)degree    |
+| Rotation X | int      | Rotation origin X position           | [centi-](https://en.wikipedia.org/wiki/Centi-)"keyunit" |
+| Rotation Y | int      | Rotation origin Y position           | [centi-](https://en.wikipedia.org/wiki/Centi-)"keyunit" |
+
+:::tip
+You can specify negative values in devicetree using parentheses around it, e.g. `(-3000)` for a 30 degree counterclockwise rotation.
+:::
+
+ZMK defines a number of popular physical layouts in-tree at [`app/dts/layouts`](https://github.com/zmkfirmware/zmk/tree/main/app/dts/layouts).
+To use such layouts, import them and assign their `transform` and (optionally) `kscan` properties.
+
+### Physical Layout with Keys Example
+
+Here is an example of a physical layout for a 2x2 macropad:
+
+```dts
+#include <physical_layouts.dtsi>
+
+/ {
+    macropad_physical_layout: macropad_physical_layout {
+        compatible = "zmk,physical-layout";
+        display-name = "Macro Pad";
+        transform = <&default_transform>;
+        kscan = <&kscan0>;
+        keys  //                     w   h    x    y     rot    rx    ry
+            = <&key_physical_attrs 100 100    0    0       0     0     0>
+            , <&key_physical_attrs 100 100  100    0       0     0     0>
+            , <&key_physical_attrs 100 100    0  100       0     0     0>
+            , <&key_physical_attrs 100 100  100  100       0     0     0>
+            ;
+    };
+};
+```
+
+### Predefined Layout Usage Example
```

Comment:
```suggestion
## Using Predefined Layouts
```
How about moving line 61 here and making this its own level 2 header?

As it is, it seemed a bit awkward to me to have this example with 4 layouts under "basic", *then* have a section on "multiple layouts"

Reply:
Good idea. I've also added a quick example using the ferris' layout, since that's not a multiple layouts situation, and a sentence to make the situation there clearer.